react/features/app/getRouteToRender.js

```diff
@@ -49,60 +49,71 @@ export function _getRouteToRender(stateful: Function | Object): Promise<Route> {
         href: undefined
     };
 
-    return new Promise(resolve => {
-        // First, check if the current endpoint supports WebRTC. We are
-        // intentionally not performing the check for mobile browsers because:
-        // - the WelcomePage is mobile ready;
-        // - if the URL points to a conference, getDeepLinkingPage will take
-        //   care of it.
-        if (!isMobileBrowser && !JitsiMeetJS.isWebRtcSupported()) {
-            route.component = UnsupportedDesktopBrowser;
-            resolve(route);
+    const joiningValidRoom = isRoomValid(room);
 
-            return;
+    // Mobile custom routing
+    if (isMobileApp) {
+        if (joiningValidRoom) {
+            route.component = Conference;
+        } else if (isWelcomePageAppEnabled(state)) {
+            route.component = WelcomePage;
         }
 
-        if (isRoomValid(room)) {
-            if (isMobileApp) {
-                route.component = Conference;
-                resolve(route);
-            } else {
-                // Update the location if it doesn't match. This happens when a
-                // room is joined from the welcome page. The reason for doing
-                // this instead of using the history API is that we want to load
-                // the config.js which takes the room into account.
-                const { locationURL } = state['features/base/connection'];
-
-                // eslint-disable-next-line no-negated-condition
-                if (window.location.href !== locationURL.href) {
-                    route.href = locationURL.href;
-                    resolve(route);
-                } else {
-                    // Maybe show deep-linking, otherwise go to Conference.
-                    getDeepLinkingPage(state).then(component => {
-                        route.component = component || Conference;
-                        resolve(route);
-                    });
-                }
-            }
+        return Promise.resolve(route);
+    }
```

Comment:
if it's a mobile app and the room isn't valid and `isWelcomePageAppEnabled` is false, should mobile fall through to the web logic?

Reply:
Web has special handling that I think mobile does not have, which is when I decided not to mix the two platforms together. So the answer is no, at the least it should not have the same logic with deep linking and web has its own welcome page checks.

Reply:
That said, please do look at what's in current master and let me know if I missed any cases. For example, Pawel had to tell me about a case I missed on web which is for unsupported browsers not to show the welcome page.